src/config.js

```diff
@@ -589,8 +594,18 @@ class Config {
       [keyPath] = args
     }
 
+    const noSources = !options.sources || (Array.isArray(options.sources) && options.sources.length === 0)
+    if (this.hasCurrentProject && noSources) {
+      const projectScope = Array.isArray(scope) ? scope.push(PROJECT) : [PROJECT]
+      const projectOptions = Object.assign({sources: [PROJECT]}, options)
+      value = this.getRawScopedValue(projectScope, keyPath, projectOptions)
+      if (value != null) {
+        return value
+      }
+    }
```

Comment:
Can you explain why this logic :point_up: (and also the additions to `getAll` below) is needed?

I would think that `get` and `getAll` would not need to change much if at all; we'd just need to give the `PROJECT` source a higher priority, kind of like what [we do for the user settings](https://github.com/atom/atom/blob/7b490c66ec6a9b277c44db082383ea1c5f50a2f3/src/config.js#L1170-L1172).